pkg/controller/storagecluster/stork.go

```diff
@@ -787,6 +787,21 @@ func (c *Controller) getStorkDeploymentSpec(
 								},
 							},
 						},
+						NodeAffinity: &v1.NodeAffinity{
```

Comment:
We don't need to add the affinity here if it is added to the storagecluster. Check line 827 to see that we copy the node affinity from the stc to this deployment. 

Same for the stork scheduler below.

Reply:
removed the affinity from all component, and added affinity in **getSelectorTerms**